synapse/push/httppusher.py

```diff
@@ -333,10 +333,10 @@ def dispatch_push(self, event, tweaks, badge):
             defer.returnValue([])
         try:
             resp = yield self.http_client.post_json_get_json(self.url, notification_dict)
-        except Exception:
-            logger.warn(
-                "Failed to push event %s to %s",
-                event.event_id, self.name, exc_info=True,
+        except Exception as e:
+            logger.warning(
+                "Failed to push event %s to %s: %s %s",
+                event.event_id, self.name, type(e), e,
```

Comment:
I'm a bit cautious about catching all the exceptions here in case it masks actual programming exceptions, can we not catch `HttpResponseException`  etc?

OTOH, it was already logging at warning, so this isn't the end of the world

Reply:
wellll....

normally I'd agree with you, but the exception that was causing a problem is a `twisted.web._newclient.RequestGenerationFailed: [<twisted.python.failure.Failure twisted.internet.defer.CancelledError: >]`. AFAICT that doesn't seem to be part of twisted's public interface, so it seems to be not a great idea to try and catch it here.

Reply:
Oh great. It also doesn't look like I added `RequestSendFailed` stuff to the normal http client either.